### ISSUE TITLE
[NEXT-1220] layout.js rendering twice in production

### ISSUE THREAD

**Issue by joshwcomeau** [vercel/next.js#49115]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
      Platform: darwin
      Arch: arm64
      Version: Darwin Kernel Version 22.4.0: Mon Mar  6 20:59:28 PST 2023; root:xnu-8796.101.5~3/RELEASE_ARM64_T6000
    Binaries:
      Node: 16.17.0
      npm: 8.15.0
      Yarn: 1.22.19
      pnpm: N/A
    Relevant packages:
      next: 13.3.5-canary.3
      eslint-config-next: 13.3.4
      react: 18.2.0
      react-dom: 18.2.0
```

### Which area(s) of Next.js are affected? (leave empty if unsure)

App directory (appDir: true)

### Link to the code that reproduces this issue

https://codesandbox.io/p/sandbox/layout-render-twice-u24kxo

### To Reproduce

- Visit the CodeSandbox link
- Open the "dev:logs" pane
- Notice that duplicate logs occur:
<img width="236" alt="CleanShot 2023-05-02 at 18 14 55@2x" src="https://user-images.githubusercontent.com/6692932/235797472-2e220ece-e9c6-4e13-84cf-f96a4656b09d.png">

### Describe the Bug

The root `layout.js` will render twice on the server for every request, when using dynamic routing (`const dynamic = 'force-dynamic';`). This happens even though it's a Server Component (no `"use client"` directive).

My first thought was that this was Strict Mode related, but the double-render occurs even when disabling Strict Mode in `next.config.js`, as well as when serving a production build.

This is happening in the root `layout.js`, as well as any components rendered within `layout.js`. It does not seem to be happening in `page.js`.

### Expected Behavior

Each request should generate a single render, so that any expensive operations (eg. database calls) only fire once.

EDIT: with further testing, I notice that these two renders are done in parallel, so it wouldn't actually slow the response, but it does cause problems if there are side effects (eg. a hit counter that registers every render).

<sub>[NEXT-1220](https://linear.app/vercel/issue/NEXT-1220/layoutjs-rendering-twice-in-production)</sub>

**Comment by timneutkens** [comment]
Hey Josh, I can't open the sandbox, happy to take a look. Without looking at the code I'm assuming what you're seeing is the preloading behavior, which will only end up with a single render but it's not guaranteed that React only ever renders once anyway because of Suspense boundaries.

> Each request should generate a single render, so that any expensive operations (eg. database calls) only fire once.

All expensive operations should leverage `import { cache } from 'react'` so that they only ever happen once per request instead of once per call, that way rerendering because of e.g. a Suspense boundary resolving would cause a similar second render.

**Comment by joshwcomeau** [comment]
Hi @timneutkens! Ahh that's super helpful, thanks for the context!

In case it's useful, here's the code from that codesandbox, in the root `layout.js`:

```jsx
export default function RootLayout({ children }) {
  console.log("Render", Date.now());
  
  return (
    <html lang="en">
      <body>{children}</body>
    </html>
  );
}
```

I see two logs firing 1-2 milliseconds apart when viewing the root page.

RE: preloading behavior, do you happen to know where I can learn more about it? Tried googling but I couldn't find anything. Although I realize all this stuff is in beta / docs might not exist yet.

**Comment by choisuill** [comment]
I have same issue
Why is it rendered twice?

`
// app / layout.tsx
export default async function RootLayout({ children }: IProps) {
  console.log('RootLayout Render');
  return (
    <html lang="en">
      <body>
         <main>{children}</main>
      </body>
    </html>
  );
}`

`
// app / page.tsx
export default async function Home() {
  console.log('Home Render');
  return <div>Home</div>
}
`

`
result
Home Render
RootLayout Render
RootLayout Render
`

**Comment by mohamadfala** [comment]
> I have same issue Why is it rendered twice?
> 
> ```
> // app / layout.tsx
> export default async function RootLayout({ children }: IProps) {
>   console.log('RootLayout Render');
>   return (
>     <html lang="en">
>       <body>
>          <main>{children}</main>
>       </body>
>     </html>
>   );
> }`
> ```
> 
> ```
> // app / page.tsx
> export default async function Home() {
>   console.log('Home Render');
>   return <div>Home</div>
> }
> ```
> 
> ```
> result
> Home Render
> RootLayout Render
> RootLayout Render
> ```

I'm facing the exact same issue! i tried removing everything and kept it as basic code exactly as you provided, but it keeps rendering twice (not because of strict mode, i have turned this off already).

I'm using next-auth, but i did remove the session provider and everything else that could affect this behavior.

any luck finding a resolution to your issue?

**Comment by mohamadfala** [comment]
@joshwcomeau I was wondering if you did find your issue.. 👀

**Comment by timneutkens** [comment]
This is caused by the root not-found boundary being applied by default, so it renders the layout around the not-found boundary in order for it to render correctly when notFound() is called in the root layout under certain conditions. This can be improved in the future so I've synced it into our issue tracker.

**Comment by 0xAnakin** [comment]
Any updates on this?

**Comment by Stewcha** [comment]
still nothing, i have the same bug?

**Comment by karlhorky** [comment]
I think the PR that fixed this (#52589) may have removed the layout in our `not-found.tsx` page:

Copying [my comment](https://github.com/vercel/next.js/pull/52589#issuecomment-1635633871) over here if it's helpful for anyone:

---

Hmm... wonder if this is causing the layout to be missing on the `not-found.tsx` page:

Before (`next@13.4.9`, left navigation from layout):

![not-found](https://github.com/vercel/next.js/assets/1935696/f68bd44b-8953-4958-bbb8-e4a8568d925f)


After (`next@13.4.10`, no left navigation from layout):

![not-found](https://github.com/vercel/next.js/assets/1935696/43992bb8-460c-4436-aa1a-e25dc0385224)

**Comment by karlhorky** [comment]
Opened an issue to report the problem of `not-found` losing the layout here:

- https://github.com/vercel/next.js/issues/52718
